### PR TITLE
Verify adminEmails against sessionID in saveword endpoint

### PR DESCRIPTION
**Pull request by arminvoid**

### CODE REVIEW

File: routes/adminapi.js [hunk 1]
```diff
@@ -61,23 +61,27 @@ router.post('/saveword', function(req, res) {
     if (!(config.COOKIE_NAME in req.cookies))
```
Comment (prateekbh) [inline routes/adminapi.js]: config.SAVEWORD_KEY_VALIDATION is needed any longer?

File: routes/adminapi.js [hunk 2]
```diff
@@ -61,23 +61,27 @@ router.post('/saveword', function(req, res) {
...
       return res.status(400).send('Cookie not provided.');
     var sessionId = req.cookies[config.COOKIE_NAME];
+    db.getUser(sessionId, (user) => {
+      if (user === false || user === 'No User.') return res.status(500).send('Error Occurred');
```
Comment (prateekbh) [inline routes/adminapi.js]: yeah so basically this works, but i guess you must have realised by now how ugly this if/else checks looks.
may be we can come back to this later and convert this to `Promises`

File: routes/adminapi.js [hunk 3]
```diff
@@ -61,23 +61,27 @@ router.post('/saveword', function(req, res) {
...
+      if (config.WHITELISTED_ADMINS.indexOf(user.email) < 0) return res.status(400);
+      let status = {
+        'Saved': 200,
+        'Duplicate': 400,
+        'Error': 500,
+      };
+      // validate(req.body)
+      db.checkWordExists(req.body.word, (message) => {
```
Comment (prateekbh) [inline routes/adminapi.js]: should we record the author of the word?

Comment (prateekbh) [reply routes/adminapi.js]: if its not too much

File: routes/userapp.js [hunk 4]
```diff
@@ -22,7 +22,9 @@ OR
 router.post('/login', function(req, res, next) {
   // Check if auth_token is provided
   if (req.body.hasOwnProperty(config.AUTH_TOKEN)) {
-    getProfileFromGoogle(req.body[config.AUTH_TOKEN], (googleName, email) => {
+    getProfileFromGoogle(req.body[config.AUTH_TOKEN],
+      (err, googleName, email) => {
```
Comment (prateekbh) [inline routes/userapp.js:26]: use named params 
http://stackoverflow.com/questions/11796093/named-parameters-in-javascript

File: routes/userapp.js [hunk 5]
```diff
@@ -52,9 +54,9 @@ let loginResponse = function(_id, name, res) {
 let getProfileFromGoogle = function(accessToken, callback) {
   const credential = firebase.auth.GoogleAuthProvider.credential(accessToken);
   firebase.auth().signInWithCredential(credential).then((result)=>{
-    callback(result.displayName, result.email);
+    callback(null, result.displayName, result.email);
```
Comment (prateekbh) [inline routes/userapp.js:57]: with named params, you wont need this

**Review by prateekbh — COMMENTED** [review]
merge latest master and push

File: routes/adminapi.js (hunk 1 above)
Comment (arminvoid) [reply routes/adminapi.js]: Removed.

File: routes/adminapi.js (hunk 2 above)
Comment (arminvoid) [reply routes/adminapi.js]: Most definitely.

File: routes/adminapi.js (hunk 3 above)
Comment (arminvoid) [reply routes/adminapi.js]: Done.

File: routes/adminapi.js [hunk 6]
```diff
@@ -54,29 +54,33 @@ router.get('/search', (req, response, next) => {
...
-    'Duplicate': 400,
-    'Error': 500,
-  };
-  // validate(req.body)
-  db.checkWordExists(req.body.word, (message) => {
-    if (message === false) {
-      db.insertWordInCollection(req.body, (message) => {
+  if (!(config.COOKIE_NAME in req.cookies))
+  return res.status(400).send('Cookie not provided.');
+  var sessionId = req.cookies[config.COOKIE_NAME];
```
Comment (prateekbh) [inline routes/adminapi.js:63]: just a note:
bro let/const are the new age var..
they also give your lexical scope vs functional scope of var

**Pull request merged**